great_expectations/core/batch.py

```diff
@@ -871,50 +871,39 @@ def get_batch_request_from_acceptable_arguments(
 def standardize_batch_request_display_ordering(
     batch_request: Dict[str, Union[str, int, Dict[str, Any]]]
 ) -> Dict[str, Union[str, Dict[str, Any]]]:
-    datasource_name: str = batch_request["datasource_name"]
-    data_connector_name: str = batch_request["data_connector_name"]
-    data_asset_name: str = batch_request["data_asset_name"]
+    datasource_name: str = batch_request.get("datasource_name")
+    data_connector_name: str = batch_request.get("data_connector_name")
+    data_asset_name: str = batch_request.get("data_asset_name")
     runtime_parameters: str = batch_request.get("runtime_parameters")
     batch_identifiers: str = batch_request.get("batch_identifiers")
-    batch_request.pop("datasource_name")
-    batch_request.pop("data_connector_name")
-    batch_request.pop("data_asset_name")
-    # NOTE: AJB 20211217 The below conditionals should be refactored
```

Comment:
Thanks for the tip @anthonyburdi 😄

Reply:
Could you explain this a bit more? How is ordering possible in a regular dictionary?

Reply:
As of Python 3.6+ dictionaries are [insertion ordered](https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6), so this order should be preserved for display.

Reply:
I think the hash map/dictionary abstract class does not define order as part of its explicit contract to the user. While Python's implementation does indeed provide that, I think there's an argument to be made to use an `OrderedDict` or write a comment about insertion ordered.

Reply:
I changed this to use an `OrderedDict` and I believe that should take care of the confusion.